drivers/cache/cache_nrf.c

```diff
@@ -17,7 +17,9 @@ LOG_MODULE_REGISTER(cache_nrfx, CONFIG_CACHE_LOG_LEVEL);
 #define CACHE_LINE_SIZE		     32
 #define CACHE_BUSY_RETRY_INTERVAL_US 10
 
-static struct k_spinlock lock;
+/* Some SOCs has a bug that requires to set 28th bit in the address on Trustzone secure builds. */
+#define ADDR_PATCH COND_CODE_1(CONFIG_CACHE_NRF_PATCH_LINEADDR, \
+		(COND_CODE_1(CONFIG_TRUSTED_EXECUTION_NONSECURE, (0), (BIT(28)))), (0))
```

Comment:
Wouldn't it be more readable to just have in that affected function:
```
	if (IS_ENABLED(CONFIG_CACHE_NRF_PATCH_LINEADDR) &&
	    !IS_ENABLED(CONFIG_TRUSTED_EXECUTION_NONSECURE)) {
		addr |= BIT(28);
	}
```
and the above comment.

Reply:
done